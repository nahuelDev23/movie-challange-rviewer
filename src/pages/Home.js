import React from 'react'
import { Categories } from '../components/Categories'

import { HomeLayout } from '../components/layout/HomeLayout'

export const Home = () => {
    return (
        <HomeLayout>
            <Categories />
        </HomeLayout>
    )
}
