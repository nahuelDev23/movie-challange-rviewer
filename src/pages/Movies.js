import React, { useEffect, useContext } from 'react'
import { Filter } from '../components/filter/Filter'
import { MainLayout } from '../components/layout/MainLayout'
import { DataContext } from '../context/data/DataContext'

export const Movies = () => {
  const { movies, onLoadData } = useContext(DataContext)

  useEffect(() => {
    onLoadData('movie')
  }, [])
  return (
    <MainLayout>
      <Filter category='movie' />
      <h1>movies</h1>
      {
        movies && movies.map(serie => (
          <div key={serie.title}>{serie.title}</div>
        ))
      }
    </MainLayout>
  )
}
