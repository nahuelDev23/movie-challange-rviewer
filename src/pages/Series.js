import React from 'react'
import { useContext } from 'react'
import { useEffect } from 'react'
import { Filter } from '../components/filter/Filter'
import { MainLayout } from '../components/layout/MainLayout'
import { DataContext } from '../context/data/DataContext'

export const Series = () => {

  const { series, onLoadData } = useContext(DataContext)

  useEffect(() => {
    onLoadData('series')
  }, [])

  return (
    <MainLayout>
      <Filter category='series' />
      <h1>Popular Series</h1>
      {
        series && series.map(serie => (
          <div key={serie.title}>{serie.title}</div>
        ))
      }
    </MainLayout>
  )
}
