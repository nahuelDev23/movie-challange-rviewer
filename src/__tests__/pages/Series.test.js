import { shallow } from 'enzyme'
import { Series } from '../../pages/Series';

describe('Pruebas en <Series/>', () => {
    test('Debe mostrar <Series/> correctamente', () => {
        const wrapper = shallow(<Series />);
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe hacer uso del MainLayout', () => {
        const wrapper = shallow(<Series />);

        const MainLayout = wrapper.find('MainLayout');
        expect(MainLayout).toBeTruthy();
    })
    test('Debe Contener el componente Filter con la propiedad series', () => {
        const wrapper = shallow(<Series />);
        const Filter = wrapper.find('Filter');
        expect(Filter.props().category).toBe('series');
    })


    test('Debe Contener el titulo Popular Series', () => {
        const wrapper = shallow(<Series />);

        const title = wrapper.find('h1').text();
        expect(title).toBe('Popular Series');
    })

    test('Debe hacer uso del componente ProductList y pasar el atributo series', () => {
        const wrapper = shallow(<Series />);

        const ProductList = wrapper.find('ProductList');
        expect(ProductList.props().type).toBe('series');
        expect(ProductList).toBeTruthy();
    })
})