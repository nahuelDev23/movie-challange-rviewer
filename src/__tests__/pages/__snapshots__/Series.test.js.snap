// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Series/> Debe mostrar <Series/> correctamente 1`] = `
<MainLayout>
  <Filter
    category="series"
  />
  <div
    className="center-content"
  >
    <div
      className="pt-4 pb-4"
    >
      <h1
        className="title"
      >
        Popular Series
      </h1>
      <ProductList
        type="series"
      />
    </div>
  </div>
</MainLayout>
`;
